REDCapUITweaker.php

```diff
@@ -1203,6 +1275,54 @@ function provideSQLDescriptive( $listFields )
 
 
 
+	// Output the current user role name
+
+	function provideUserRoleName()
+	{
+		if ( !defined( 'USERID' ) || USERID == '' )
+		{
+			return;
+		}
+
+		$userRights = $this->getUser()->getRights();
+		if ( !isset( $userRights ) ||
+		     !isset( $userRights['role_name'] ) || $userRights['role_name'] == '' )
+		{
+			return;
+		}
+
+		$roleName = $this->escapeHTML( $userRights['role_name'] );
```

Comment:
I am not sure if escapeHTML is necessary as it's not in html text in DB.   You can have this code if you want but I think it's unnecessary.

Reply:
If the escapeHTML function is not used, the module would fail the 'taint checking' applied upon submission to the REDCap module repo.

Reply:
I still don't understand how module will fail on 'taint checking' as it's user right not data entry form, also it's a reading data not variable in web.  You may show me how it failed next week.  I don't think it's harm to put this code in anyway.

Reply:
The role name might be safe, but the 'taint checking' is fairly strict and there's no harm in being over-cautious here.